spec/datadog/open_feature/evaluation_engine_spec.rb

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'datadog/open_feature/evaluation_engine'
+
+RSpec.describe Datadog::OpenFeature::EvaluationEngine do
+  let(:engine) { described_class.new(reporter, telemetry: telemetry, logger: logger) }
+  let(:reporter) { instance_double(Datadog::OpenFeature::Exposures::Reporter) }
+  let(:telemetry) { instance_double(Datadog::Core::Telemetry::Component) }
+  let(:logger) { instance_double(Datadog::Core::Logger) }
+  let(:ufc) do
+    <<~JSON
+      {
+        "data": {
+          "type": "universal-flag-configuration",
+          "id": "1",
+          "attributes": {
+            "createdAt": "2024-04-17T19:40:53.716Z",
+            "format": "SERVER",
+            "environment": { "name": "test" },
+            "flags": {
+              "test_flag": {
+                "key": "test",
+                "enabled": true,
+                "variationType": "STRING",
+                "variations": {
+                  "control": { "key": "control", "value": "hello" }
+                },
+                "allocations": [
+                  {
+                    "key": "rollout",
+                    "splits": [{ "variationKey": "control", "shards": [] }],
+                    "doLog": false
+                  }
+                ]
+              }
+            }
+          }
+        }
+      }
+    JSON
+  end
+
+  describe '#fetch_value' do
+    let(:result) { engine.fetch_value(flag_key: 'test', expected_type: :string) }
```

Comment:
```suggestion
    let(:result) { engine.fetch_value(flag_key: 'test_flag', expected_type: :string) }
```

Or update the flag keys in the example UFC to "test"

```
            "flags": {
              "test_flag": { <--
```